### DISCUSSION TITLE
-all-manifests parameter

### DISCUSSION THREAD

**Discussion by FatalBulletHit** [SteamRE/DepotDownloader#219] [Q&A]
As mentioned in #148 and #205, an `-all-manifests` parameter would be a nice addition.

**Comment by azuisleet** [comment]
There is no such hypothetical "-all-manifests". Steam only provides the current manifest for a given app and its depots.

**Reply by FatalBulletHit** [reply]
Could we not use SteamDB?

**Reply by psychonic** [reply]
SteamDB does not support programmatic access.

**Reply by PaulNevada** [reply]
> SteamDB does not support programmatic access.

Even trivial parsing not allowed?

**Reply by psychonic** [reply]
No.

**Reply by FatalBulletHit** [reply]
Even if we keep calls to a minimum, e.g. by only requesting the entire manifest history if there was an update and caching the result? 
Updates to a depot could be detected by comparing the latest manifest (which we get directly from Steam) with the latest cached one.

Ideally uploading the cached history rather than only saving it locally.

**Reply by psychonic** [reply]
Caching history of what DepotDownloader sees from Steam is a possibility. Scraping SteamDB is not.

**Reply by FatalBulletHit** [reply]
Obviously, I understand why we don't want any additional traffic on SteamDB, but I've probably caused more traffic by manually opening depot histories in my browser (which is allowed) than if I had scraped SteamDB on depot updates (which isn't allowed).
So, I think it'd also be in SteamDB's interest to allow minimum scraping as it would reduce traffic overall.

**Reply by psychonic** [reply]
Sure, if they say that scraping is allowed or provide an API, it can be considered.
